### PR TITLE
DSND-2872: Update structured logging

### PR DESCRIPTION
**Pull request by sthompsonCH**
## Describe the changes
This PR primarily adds entity id to the log map at the PUT/DELETE requests in the controller so this field can be tracked more easily, by FESS, back to CHIPS.

Other changes also made to simplify log messages.

### Related Jira tickets

[DSND-2872](https://companieshouse.atlassian.net/browse/DSND-2872)

### General

- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing

- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation

_Where possible, add links to the respective Jira tickets when an item is checked_

- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to deployment repositories?

## Notes to the tester

_Add any testing hints or instructions here._

[DSND-2872]: https://companieshouse.atlassian.net/browse/DSND-2872?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryController.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public ResponseEntity<Void> upsertFilingHistoryTransaction(
     @DeleteMapping("/filing-history/{entity_id}/internal")
     public ResponseEntity<Void> deleteFilingHistoryTransaction(@PathVariable("entity_id") final String entityId) {
 
-        DataMapHolder.get().transactionId(entityId);
+        DataMapHolder.get().entityId(entityId);
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryController.java:104]: Line 123 below, the transactionId can be removed from the message string as its in the log map

File: src/main/java/uk/gov/companieshouse/filinghistory/api/repository/Repository.java [hunk 2]
```diff
@@ -102,7 +102,7 @@ public Optional<FilingHistoryDocument> findByIdAndCompanyNumber(final String id,
 
             return Optional.ofNullable(mongoTemplate.findOne(query, FilingHistoryDocument.class));
         } catch (DataAccessException ex) {
-            LOGGER.error("MongoDB error when finding the document: %s".formatted(ex.getMessage()), ex,
+            LOGGER.error("MongoDB error when finding document: %s".formatted(ex.getMessage()), ex,
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/repository/Repository.java:105]: We need to expand the catch block into multiple catches for specific exceptions, currently its difficult to figure out what when wrong and how important it its.

**Review by crowlandsCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by cgregoryCH — APPROVED** [review]

**Review by johncook-ch — APPROVED** [review]

**Pull request merged**